### PR TITLE
Update/simplify the okhttp instrumentation based on new capabilities …

### PR DESCRIPTION
**Pull request by jkwatson**
…in the OpenTelemetry library instrumentation.

### CODE REVIEW

File: splunk-otel-android/src/test/java/com/splunk/rum/RumResponseAttributesExtractorTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        RumResponseAttributesExtractor attributesExtractor = new RumResponseAttributesExtractor(headerParser);
+        AttributesBuilder attributesBuilder = Attributes.builder();
+        attributesExtractor.onEnd(attributesBuilder, fakeRequest, response);
+        attributesExtractor.onStart(attributesBuilder, fakeRequest);
+        Attributes attributes = attributesBuilder.build();
+
+        assertEquals("http", attributes.get(SplunkRum.COMPONENT_KEY));
+        assertNull(attributes.get(OkHttpRumInterceptor.LINK_TRACE_ID_KEY));
+        assertNull(attributes.get(OkHttpRumInterceptor.LINK_SPAN_ID_KEY));
```
Comment (breedx-splk) [inline splunk-otel-android/src/test/java/com/splunk/rum/RumResponseAttributesExtractorTest.java:97]: Nit: Might be worth also verifying that the content length is still recorded even without this special header?

**Review by breedx-splk — APPROVED** [review]
Looks good. One small thought about the last test.

**Comment by breedx-splk** [comment]
Also these new APIs are 🔥

File: splunk-otel-android/src/test/java/com/splunk/rum/RumResponseAttributesExtractorTest.java (hunk 1 above)
Comment (jkwatson) [reply splunk-otel-android/src/test/java/com/splunk/rum/RumResponseAttributesExtractorTest.java:83]: added. good call

**Pull request merged**